### PR TITLE
Limit nad depth

### PR DESCRIPTION
**Pull request by FranckLecuyer**

### CODE REVIEW

File: src/components/diagrams/diagram-footer.js [hunk 1]
```diff
@@ -20,16 +21,22 @@ const useStyles = makeStyles((theme) => ({
...
     decrementCounterIcon: {
-        bottom: theme.spacing(1),
-        left: theme.spacing(2),
+        bottom: theme.spacing(0),
+        left: theme.spacing(0),
         position: 'absolute',
         cursor: 'pointer',
+        '&:hover': {
+            backgroundColor: 'transparent',
+        },
```
Comment (EstherDarkish) [inline src/components/diagrams/diagram-footer.js:39]: Use these CSS rules instead :

`    incrementCounterIcon: {`
`        padding: 0,`
`        bottom: theme.spacing(1),`
`        left: theme.spacing(5.5),`
`        position: 'absolute',`
`        cursor: 'pointer',`
`    },`
`    decrementCounterIcon: {`
`        padding: 0,`
`        bottom: theme.spacing(1),`
`        left: theme.spacing(2),`
`        position: 'absolute',`
`        cursor: 'pointer',`
`    },`

File: src/components/diagrams/diagram-footer.js [hunk 2]
```diff
@@ -74,14 +81,20 @@ const DiagramFooter = (props) => {
...
+                        disabled={props.incrementCounterDisabled}
                         className={classes.incrementCounterIcon}
-                    />
-                    <RemoveCircleIcon
+                    >
+                        <AddCircleIcon />
+                    </IconButton>
+                    <IconButton
                         onClick={handleDecrementCounter}
+                        disabled={props.decrementCounterDisabled}
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/components/diagrams/diagram-footer.js:89]: Done

File: src/redux/actions.js [hunk 3]
```diff
@@ -559,6 +559,16 @@ export function decrementNetworkAreaDiagramDepth() {
     };
 }
 
+export const NETWORK_AREA_DIAGRAM_NB_VOLTAGE_LEVELS =
+    'NETWORK_AREA_DIAGRAM_NB_VOLTAGE_LEVELS';
+
+export function networkAreaDiagramNbVoltageLevels(nbVoltageLevels) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/redux/actions.js]: Done

File: src/components/diagrams/diagram-footer.js (hunk 1 above)
Comment (FranckLecuyer) [reply src/components/diagrams/diagram-footer.js]: Done

**Review by EstherDarkish — APPROVED** [review]

File: src/components/diagrams/diagram-common.js [hunk 4]
```diff
@@ -31,6 +31,10 @@ export const MAX_WIDTH_SUBSTATION = 1200;
 export const MAX_HEIGHT_SUBSTATION = 700;
 export const MAX_WIDTH_NETWORK_AREA_DIAGRAM = 1200;
 export const MAX_HEIGHT_NETWORK_AREA_DIAGRAM = 650;
+export const NETWORK_AREA_DIAGRAM_NB_MAX_VOLTAGE_LEVELS = 200;
+
+// To allow controls that are in the corners of the map to not be hidden in normal mode
+// (but they are still hidden in fullscreen mode)
```
Comment (EstherDarkish) [inline src/components/diagrams/diagram-common.js:37]: `// To allow controls that are in the corners of the map to not be hidden in normal mode`
`// (but they are still hidden in fullscreen mode)`

These comments should not be there anymore, they were removed in a recent merge.

File: src/components/diagrams/diagram-common.js [hunk 5]
```diff
@@ -31,6 +31,10 @@ export const MAX_WIDTH_SUBSTATION = 1200;
 export const MAX_HEIGHT_SUBSTATION = 700;
 export const MAX_WIDTH_NETWORK_AREA_DIAGRAM = 1200;
 export const MAX_HEIGHT_NETWORK_AREA_DIAGRAM = 650;
+export const NETWORK_AREA_DIAGRAM_NB_MAX_VOLTAGE_LEVELS = 200;
```
Comment (EstherDarkish) [inline src/components/diagrams/diagram-common.js:34]: Maybe add a comment to explain what this constant does ?

File: src/components/diagrams/diagram-common.js (hunk 4 above)
Comment (FranckLecuyer) [reply src/components/diagrams/diagram-common.js]: done

File: src/components/diagrams/diagram-common.js (hunk 5 above)
Comment (FranckLecuyer) [reply src/components/diagrams/diagram-common.js:35]: Done

**Pull request merged**